### ISSUE TITLE
Extension matching in READERS setting

### ISSUE THREAD

**Issue by machow** [getpelican/pelican#1946]
The docs mention the READERS config may be set to a dictionary to skip processing files. For example, 

```
READERS = {'Rmd': None}
```

In this case, the behavior I expected was that Pelican would not try to process `somefile.Rmd`. However, I'm still getting the errors, "Pelican does not know how to parse %s', u'somefile.Rmd'".

Looking at the [this bit of source](https://github.com/getpelican/pelican/blob/58904af92c845309008989115df6d9de40533b2d/pelican/readers.py#L454-L461) for the Readers class, the problem seems to be that the "None Type" readers from the config will be added to self.reader_classes, but not self.readers. As far as I can tell, reader_classes isn't used beyond that point.

Maybe I missed something, though--because this behavior seems to show up in the unit tests! (I just glanced at it, so could be wrong).

**Edit**
I should also mention that for some reason Pelican is trying to parse .Rmd files in my articles folder. Same with if I change the extension to .RRRRRmd, but not if I change it to .Rme, so I wonder if there is some *md matching going on somewhere.

**Comment by avaris** [comment]
Your second observation is the correct one. [Extensions are matched with `endswith`](https://github.com/getpelican/pelican/blob/master/pelican/generators.py#L115), and they don't include `.`. As long as there is a reader for `md` files (and by default there is), pelican will try to read any `*md` file as well. So, it's a bug.

**Comment by leotrs** [comment]
@avaris would it be enough to do 
```
ext = os.path.splitext(file)[1][1:]
if extensions is False or ext in extensions:
    return True
```
?

If so, I can prepare the PR.

**Comment by justinmayer** [comment]
Addressed via 6cadbdf.
